### DISCUSSION TITLE
Disconnected graphs

### DISCUSSION THREAD

**Discussion by mdanb** [pyg-team/pytorch_geometric#2215] [General]
Both in general and specifically in Pytorch geometric, do graph convolutions require graphs to be connected? I’m asking because based on my understanding it does not, but I read somewhere that it does.

**Comment by rusty1s** [comment]
Graph Convolutions follow a simple neighborhood aggregation scheme, where node features from neighbors are transformed and aggregated to each central node. Therefore, only connected nodes will exchange information, and no information is exchanged in case nodes are disconnected.

**Reply by mdanb** [reply]
Right, but even if you had disconnected nodes, you should still be able to use them for a graph level output no?

**Reply by rusty1s** [reply]
Yes, of course. However, their feature representations will not get enhanced by any information from outside.
